### PR TITLE
feat(TPG>=6.5.0)!: promote `enable_gcfs` to GA

### PR DESCRIPTION
**Pull request by wyardley**
promote `enable_gcfs` to GA in main module

### CODE REVIEW

**Comment by wyardley** [comment]
Was the test failure related to the changes?

**Comment by apeabody** [comment]
> Was the test failure related to the changes?

I don't think so, started seeing this yesterday in several PRs:

```
Step #48 - "destroy workload-metadata-config-local":        
Step #48 - "destroy workload-metadata-config-local":        Error: Failed to parse value for host: https://
Step #48 - "destroy workload-metadata-config-local":        
Step #48 - "destroy workload-metadata-config-local":          with module.example.provider["registry.terraform.io/hashicorp/kubernetes"],
Step #48 - "destroy workload-metadata-config-local":          on ../../../examples/workload_metadata_config/main.tf line 23, in provider "kubernetes":
Step #48 - "destroy workload-metadata-config-local":          23: provider "kubernetes" {
Step #48 - "destroy workload-metadata-config-local":        
Step #48 - "destroy workload-metadata-config-local":        host must be a URL or a host:port pair: "https://"
Step #48 - "destroy workload-metadata-config-local": >>>>>> Expected process to exit with [0], but received '1'
```

**Comment by apeabody** [comment]
> > Was the test failure related to the changes?
> 
> I don't think so, started seeing this yesterday in several PRs:
> 
> ```
> Step #48 - "destroy workload-metadata-config-local":        
> Step #48 - "destroy workload-metadata-config-local":        Error: Failed to parse value for host: https://
> Step #48 - "destroy workload-metadata-config-local":        
> Step #48 - "destroy workload-metadata-config-local":          with module.example.provider["registry.terraform.io/hashicorp/kubernetes"],
> Step #48 - "destroy workload-metadata-config-local":          on ../../../examples/workload_metadata_config/main.tf line 23, in provider "kubernetes":
> Step #48 - "destroy workload-metadata-config-local":          23: provider "kubernetes" {
> Step #48 - "destroy workload-metadata-config-local":        
> Step #48 - "destroy workload-metadata-config-local":        host must be a URL or a host:port pair: "https://"
> Step #48 - "destroy workload-metadata-config-local": >>>>>> Expected process to exit with [0], but received '1'
> ```

Looks like with v6.11.0 container_cluster.endpoint is no longer populate with cluster's with private endpoints.

**Comment by wyardley** [comment]
Hmm, maybe related to https://github.com/hashicorp/terraform-provider-google/pull/20193?

**Comment by apeabody** [comment]
> Hmm, maybe related to [hashicorp/terraform-provider-google#20193](https://github.com/hashicorp/terraform-provider-google/pull/20193)?

Yeah - Looking at a potential solution: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/2180

File: autogen/main/variables.tf.tmpl [hunk 1]
```diff
@@ -614,6 +614,17 @@ variable "enable_confidential_nodes" {
...
+variable "enable_gcfs" {
+  type        = bool
+  description = "Enable image streaming on cluster level."
+  {% if autopilot_cluster != true %}
+  default     = false
+  {% endif %}
+  {% if autopilot_cluster %}
+  default     = true
+  {% endif %}
+}
```
[the remark this answers is not in the document]
Comment (apeabody) [reply autogen/main/variables.tf.tmpl:626]: this is fine :)

**Review by apeabody — APPROVED** [review]
Thanks for the contribution @wyardley!

**Pull request merged**